ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts

```diff
@@ -1438,4 +1479,52 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
     this.universalModal.show();
     return false;
   }
+
+  public getMercuryEligibleSamples(): void {
+    if (!this.canHaveSequencing( this.participant )) {
+      return;
+    }
+    const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
+    this.dsmService.getMercuryEligibleSamples( this.participant.participant.ddpParticipantId, realm ).subscribe( {
+      next: data => {
+        const jsonData = data;
+        this.sequencingOrdersArray = [];
+        if (jsonData) {
+          jsonData.forEach( ( json ) => {
+              const order = SequencingOrder.parse( json );
+              this.sequencingOrdersArray.push( order );
+            }
+          );
+        }
+
+      },
+      error: () => {
+
+      }
+    } );
+  }
+
+  canHaveSequencing( participant: Participant ): boolean {
+    if (!this.role.allowedToDoOrderSequencing() || !this.hasSequencingOrders) {
+      return false;
+    }
+    const enrolled: boolean = participant.data.status === 'ENROLLED';
+    let hasGender = false;
+    if (participant.oncHistoryDetails.find( onc => onc.gender !== '' && onc.gender !== undefined && onc.gender !== null )) {
```

Comment:
#Optional
I would take this conditional check into other function and here I would just have a function call, like this:

```
if(isTrue()) {
}
```